rewrite-java/src/main/java/org/openrewrite/java/AddScopeToInjectedClass.java

```diff
@@ -0,0 +1,111 @@
+package org.openrewrite.java;
+
+import lombok.Data;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.ScanningRecipe;
+import org.openrewrite.Tree;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.tree.*;
+
+import java.util.*;
+
+public class AddScopeToInjectedClass extends ScanningRecipe<AddScopeToInjectedClass.Accumulator> {
+    private static final Collection<String> typesPromptingScopeAddition = Arrays.asList("javax.inject.Inject");
+
+    @Override
+    public String getDisplayName() {
+        return "Add scope annotation to injected classes";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Add scope annotation to injected classes.";
+    }
+
+    private static boolean variableTypeRequiresScope(@Nullable JavaType.Variable memberVariable) {
+        if (memberVariable == null) {
+            return false;
+        }
+
+        for (JavaType.FullyQualified annotation : memberVariable.getAnnotations()) {
+            if (typesPromptingScopeAddition.contains(annotation.getFullyQualifiedName())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public Accumulator getInitialValue(ExecutionContext ctx) {
+        return new Accumulator();
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getScanner(Accumulator acc) {
+        return new TreeVisitor<Tree, ExecutionContext>() {
+            @Override
+            public @Nullable Tree visit(@Nullable Tree tree, ExecutionContext ctx) {
+                // look for classes requiring scope definition
+                if (tree instanceof J.CompilationUnit) {
+                    J.CompilationUnit cu = (J.CompilationUnit) tree;
+                    for (J.ClassDeclaration classDeclaration : cu.getClasses()) {
+                        JavaType.FullyQualified type =(JavaType.FullyQualified) classDeclaration.getType();
+                        for (JavaType.Variable variable : type.getMembers())
+                        if (variableTypeRequiresScope(variable)) {
+                            acc.getScopeTypes().add((JavaType.FullyQualified)variable.getType());
+                        }
+                    }
+                }
+                return tree;
+            }
```

Comment:
You can potentially simplify this a bit by returning a JavaIsoVisitor. Here's a quick sample.
```suggestion
        return new JavaIsoVisitor<ExecutionContext>() {
            @Override
            public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext executionContext) {
                J.ClassDeclaration cd = super.visitClassDeclaration(classDecl, executionContext);
                JavaType.FullyQualified type = cd.getType();
                for (JavaType.Variable variable : type.getMembers()) {
                    if (variableTypeRequiresScope(variable)) {
                        acc.getScopeTypes().add((JavaType.FullyQualified)variable.getType());
                    }
                }
                return cd;
            }
        };
```